dbms/src/DataTypes/DataTypeTuple.cpp

```diff
@@ -162,9 +162,8 @@ void DataTypeTuple::serializeText(const IColumn & column, size_t row_num, WriteB
     writeChar('(', ostr);
     for (const auto i : ext::range(0, ext::size(elems)))
     {
-        if (i != 0)
-            writeChar(',', ostr);
         elems[i]->serializeAsTextQuoted(extractElementColumn(column, i), row_num, ostr, settings);
+        writeChar(',', ostr);
```

Comment:
I don't sure that we always need to add an extra comma at the end of tuple, because previous versions of ClickHouse cannot parse it. And `(1,2,3)` looks better than `(1,2,3,)` as for me